### PR TITLE
Added exceptions for when CasOCSolver solve() fails in MocoCasADiSolver

### PR DESCRIPTION
**Pull request by AllisonJohn**
Fixes issue #3730

### Brief summary of changes
Instead of continuing on, MocoCasADiSolver will throw exceptions when the solver doesn't fails so that the next lines don't cause more errors.

### Testing I've completed
Tested putting opensim, casadi, and std errors in the try section to see the error messages when running the sliding mass problem. Without the added errors it works normally.

### CHANGELOG.md (choose one)

- updated.

- - -
This change is [Reviewable](https://reviewable.io/reviews/opensim-org/opensim-core/3834)

### CODE REVIEW

**Review by nickbianco — CHANGES_REQUESTED** [review]
Great @AllisonJohn! I have mostly style suggestions.

In addition to the requested changes, could you provide screenshots of what each individual error message looks like when printed to the console?

File: OpenSim/Moco/MocoCasADiSolver/MocoCasADiSolver.cpp [hunk 1]
```diff
@@ -383,10 +383,25 @@ MocoSolution MocoCasADiSolver::solveImpl() const {
...
     CasOC::Solution casSolution;
+    // catch solve errors
     try {
         casSolution = casSolver->solve(casGuess);
-    } catch (...) {
-        OpenSim::Logger::setLevel(origLoggerLevel);
+    }
+    // opensim exceptions
+    catch(const Exception& ex) {
+        OPENSIM_THROW_FRMOBJ(Exception, fmt::format("MocoCasADiSolver failed internally "
```
[the remark this answers is not in the document]
Comment (AllisonJohn) [reply OpenSim/Moco/MocoCasADiSolver/MocoCasADiSolver.cpp:392]: It fits without the , ex.getMessage())); part

[the remark this answers is not in the document]
Comment (nickbianco) [reply OpenSim/Moco/MocoCasADiSolver/MocoCasADiSolver.cpp:392]: Gotcha. Then I'd go with:

```c++
OPENSIM_THROW_FRMOBJ(Exception, 
        fmt::format("MocoCasADiSolver failed internally with message: {}" , 
                ex.getMessage()));
```

**Comment by AllisonJohn** [comment]
Here is what happens when adding each of these lines to the catch:

OPENSIM_THROW(Exception, "test opensim exception"); -> 
image

casadi_error("test casadi exception"); ->
image
"libc++abi: terminating due to uncaught exception of type OpenSim::Exception: MocoCasADiSolver failed internally with message: /Users/allisonjohn/Documents/opensim-core/OpenSim/Moco/MocoCasADiSolver/MocoCasADiSolver.cpp:388: test casadi exception
	In Object '<no-name>' of type MocoCasADiSolver.
	Thrown at MocoCasADiSolver.cpp:396 in solveImpl()."

throw std::invalid_argument("test exception"); ->
image

**Review by nickbianco — APPROVED** [review]
LGTM.

**Pull request merged**